openmc/summary.py

```diff
@@ -121,22 +85,21 @@ def _read_materials(self):
             # Set the Material's density to g/cm3 - this is what is used in OpenMC
             material.set_density(density=density, units='g/cm3')
 
-            # Add all Nuclides to the Material
-            for i, zaid in enumerate(nuclides):
-                nuclide = self.get_nuclide_by_zaid(zaid)
-                density = nuc_densities[i]
+            # Add all nuclides to the Material
+            for fullname, density in zip(nuclides, nuc_densities):
+                fullname = fullname.decode().strip()
+                name, xs = fullname.split('.')
+
+                if 'nat' in name:
+                    nuclide = openmc.Element(name=name, xs=xs)
+                else:
+                    nuclide = openmc.Nuclide(name=name, xs=xs)
```

Comment:
Rather than use ``nuclide`` for an instance of a ``Nuclide`` or an ``Element``, why not merge the following ``if-else`` statement which adds the ``Nuclide``/``Element`` to the ``Material`` into this ``if-else`` statement.